Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -397,67 +350,111 @@ def _foregroundWidths(self):
-            beamAngle = self.getProperty(Prop.BEAM_ANGLE).value
-            dbPosWS = ''
-            if not self.getProperty(Prop.DIRECT_BEAM_POS_WS).isDefault:
-                dbPosWS = self.getProperty(Prop.DIRECT_BEAM_POS_WS).value
-            filename = flattened.pop(0)
-            numor = os.path.basename(filename).split('.')[0]
-            firstWSName = self._names.withSuffix('raw-' + numor)
-            beamPosWSName = self._names.withSuffix('beamPos-' + numor)
-            LoadILLReflectometry(Filename=filename,
-                                 OutputWorkspace=firstWSName,
-                                 DirectBeamPosition=dbPosWS,
-                                 BeamCentre=beamCentre,
-                                 BraggAngle=beamAngle,
-                                 XUnit='TimeOfFlight',
-                                 OutputBeamPosition=beamPosWSName,
-                                 EnableLogging=self._subalgLogging)
-            mergedWS = mtd[firstWSName]
-            beamPosWS = mtd[beamPosWSName]
-            self._cleanup.cleanupLater(beamPosWS)
+        """Return a raw input workspace."""
+        inputFiles = self.getPropertyValue(Prop.RUN)
+        inputFiles = inputFiles.replace(',', '+')
+        if inputFiles:
             mergedWSName = self._names.withSuffix('merged')
-            for i, filename in enumerate(flattened):
-                numor = os.path.basename(filename).split('.')[0]
-                rawWSName = self._names.withSuffix('raw-' + numor)
-                LoadILLReflectometry(Filename=filename,
-                                     OutputWorkspace=rawWSName,
-                                     DirectBeamPosition=dbPosWS,
-                                     XUnit='TimeOfFlight',
-                                     EnableLogging=self._subalgLogging)
-                rawWS = mtd[rawWSName]
-                mergedWS = MergeRuns(InputWorkspaces=[mergedWS, rawWS],
-                                     OutputWorkspace=mergedWSName,
-                                     EnableLogging=self._subalgLogging)
-                if i == 0:
-                    self._cleanup.cleanup(firstWSName)
-                self._cleanup.cleanup(rawWS)
-            return mergedWS, beamPosWS
-        ws = self.getProperty(Prop.INPUT_WS).value
-        self._cleanup.protect(ws)
-        if not self.getProperty(Prop.BEAM_POS_WS).isDefault:
-            beamPosWS = self.getProperty(Prop.BEAM_POS_WS).value
-            self._cleanup.protect(beamPosWS)
+            linePosition = 127.5
+            if not self.getProperty(Prop.LINE_POSITION_INPUT):
+                linePosition = self.getProperty(Prop.LINE_POSITION_INPUT).value
+            loadOption = {
+                'XUnit': 'TimeOfFlight',
+                'BeamCentre': linePosition
+            }
+            if not self.getProperty(Prop.TWO_THETA).isDefault:
+                twoTheta = self.getProperty(Prop.TWO_THETA).value / 2.
```

Comment:
Nice that we are dividing 2theta here. However, the result should not be called `twoTheta` but rather `theta` or `braggAngle` or similar.

Reply:
theta like used below would be good